src/lro-poller.ts

```diff
@@ -0,0 +1,78 @@
+import { Queue } from "./throttler/queue";
+import * as api from "./api";
+import * as FirebaseError from "./error";
+
+export interface LroPollerOptions {
+  pollerName?: string;
+  apiOrigin: string;
+  apiVersion: "v1" | "v1beta1";
+  operationResourceName: string;
+  backoff?: number;
+  masterTimeout?: number;
+}
+
+export interface LroResult {
```

Comment:
Instead of having a result that either contains an error or a response, it seems like the promise should just resolve with a response or reject with an error.

Reply:
I also agree with this. I have some code that uses the `poller` now and I have to check for `response.error` to throw the error again.

Reply:
Removed `LroResult`. The method now returns the response when the LRO is `done` with a `response`, and reject with an error in all other cases